pkg/sysregistriesv2/system_registries_v2.go

```diff
@@ -278,11 +278,24 @@ func GetRegistries(ctx *types.SystemContext) ([]Registry, error) {
 	configPath := getConfigPath(ctx)
 
 	configMutex.Lock()
-	defer configMutex.Unlock()
 	// if the config has already been loaded, return the cached registries
 	if registries, inCache := configCache[configPath]; inCache {
+		configMutex.Unlock()
 		return registries, nil
 	}
+	configMutex.Unlock()
+
+	return GetRegistriesUncached(ctx)
+}
+
+// GetRegistriesUncached loads the registries from the provided `SystemContext`
+// without using the internal cache. On success, the found registries will be
+// memoized into the internal registry cache.
```

Comment:
nit;  s/memoized/memorized/ or perhaps "added"

Reply:
https://en.wikipedia.org/wiki/Memoization ; though maybe that indicates that using simpler words would be easier to understand.

Reply:
Yes, that's what I meant, but I changed to to `added`.